pkg/spire/spire_gke.go

```diff
@@ -0,0 +1,58 @@
+package spire
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	"github.com/spiffe/go-spiffe/v2/spiffetls/tlsconfig"
+	"github.com/spiffe/go-spiffe/v2/workloadapi"
+	"io/fs"
+	"os"
+)
+
+const SPIFFE_ENDPT = "SPIFFE_ENDPOINT_SOCKET"
+
+func HasSpireEndpoint() bool {
+	s := os.Getenv(SPIFFE_ENDPT)
+	if s == "" {
+		return false
+	}
+	fi, err := os.Stat(s)
+	if err != nil {
+		return false
+	}
+	if fi.Mode().Type()&fs.ModeSocket != 0 {
+		return true
+	}
+	return false
+}
+
+func GetTlsClientConfig() (*tls.Config, error) {
```

Comment:
```suggestion
func GetTLSClientConfig() (*tls.Config, error) {
```
will cause a linting error.
> https://github.com/golang/lint/blob/master/lint.go#L795